apps/native-component-list/src/screens/AdMobScreen.tsx

```diff
@@ -2,59 +2,97 @@ import {
 
-export default class AdMobScreen extends React.Component {
-  static navigationOptions = {
-    title: 'AdMob',
-  };
+export default function AdMobScreen() {
+  const [isAvailable, error] = useResolvedValue(isAvailableAsync);
 
-  state = {
-    isInterstitialReady: false,
-    isRewardedReady: false,
-    servePersonalizedAds: false,
-  };
+  const warning = React.useMemo(() => {
+    if (error) {
+      return `An unknown error occurred while checking the API availability: ${error.message}`;
+    } else if (isAvailable === null) {
+      return 'Checking availability...';
+    } else if (isAvailable === false) {
+      return 'AdMob API is not available on this platform.';
+    }
+    return null;
+  }, [error, isAvailable]);
 
-  constructor(props: object) {
-    super(props);
-    setTestDeviceIDAsync('EMULATOR');
-    AdMobRewarded.setAdUnitID('ca-app-pub-3940256099942544/1033173712');
-    AdMobInterstitial.setAdUnitID('ca-app-pub-3940256099942544/1033173712');
+  if (warning) {
+    return (
+      <View style={{ justifyContent: 'center', alignItems: 'center', flex: 1 }}>
+        <Text>{warning}</Text>
+      </View>
+    );
   }
 
-  componentDidMount() {
-    AdMobRewarded.addEventListener('rewardedVideoDidClose', this.reloadRewarded);
-    AdMobInterstitial.addEventListener('interstitialDidClose', this.reloadInterstitial);
-    this.reloadRewarded();
-    this.reloadInterstitial();
-  }
+  return <AdMobView />;
+}
 
-  componentWillUnmount() {
-    AdMobRewarded.removeEventListener('rewardedVideoDidClose', this.reloadRewarded);
-    AdMobInterstitial.removeEventListener('interstitialDidClose', this.reloadInterstitial);
-  }
+const AdMobBannerTestUnitID = Platform.select({
+  // https://developers.google.com/admob/ios/test-ads
+  ios: 'ca-app-pub-3940256099942544/2934735716',
+  // https://developers.google.com/admob/android/test-ads
+  android: 'ca-app-pub-3940256099942544/6300978111',
+});
 
-  onPress = () => {
+const AdMobInterstitialTestUnitID = Platform.select({
+  // https://developers.google.com/admob/ios/test-ads
+  ios: 'ca-app-pub-3940256099942544/4411468910',
+  // https://developers.google.com/admob/android/test-ads
+  android: 'ca-app-pub-3940256099942544/1033173712',
+})!;
+
+const AdMobRewardedTestUnitID = Platform.select({
+  // https://developers.google.com/admob/ios/test-ads
+  ios: 'ca-app-pub-3940256099942544/1712485313',
+  // https://developers.google.com/admob/android/test-ads
+  android: 'ca-app-pub-3940256099942544/5224354917',
+})!;
+
+function AdMobView() {
+  const [isInterstitialReady, setInterstitialReady] = React.useState(false);
+  const [isRewardedReady, setRewardedReady] = React.useState(false);
+  const [servePersonalizedAds, setPersonalizedAds] = React.useState(false);
+
+  React.useEffect(() => {
+    setTestDeviceIDAsync('EMULATOR');
+    AdMobRewarded.setAdUnitID(AdMobRewardedTestUnitID);
+    AdMobInterstitial.setAdUnitID(AdMobInterstitialTestUnitID);
```

Comment:
Now that I think about it… isn't this a problem that these IDs are set asynchronously after the components are rendered? Maybe this should be moved to global scope of the file?

Reply:
No cuz we have to first check availability first.